### PR TITLE
API runs with gunicorn/uvicorn combo

### LINKED ISSUE #3174: API runs in "production" mode
## Description

As a TRE Administrator / Developer
I want to ensure that the API application runs with recommended settings for production use
So that it will be more stable and able to service more requests

### PR DESCRIPTION
**Pull request by tamirkamara**
Resolves #3174 

## What is being addressed

Running the API in production hasn't followed the recommended pattern as described here: https://www.uvicorn.org/deployment/#using-a-process-manager

## How is this addressed

- For deployment, when running the API docker image, use gunicorn with uvicorn worker(s). Currently the number of workers is still 1 to confirm everything works as we're used to (in the future a higher number is probably in order).

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -33,6 +33,7 @@ ENHANCEMENTS:
 * Move non-core DNS zones out of the network module to reduce dependencies [#3119](https://github.com/microsoft/AzureTRE/pull/3119)
 * Review VMs are being cleaned up when an Airlock request is canceled ([#3130](https://github.com/microsoft/AzureTRE/pull/3130))
 * Sample queries to investigate logs of the core TRE applications ([#3151](https://github.com/microsoft/AzureTRE/pull/3151))
+* API runs with gunicorn and uvicorn workers (as recommanded) [#TBD](https://github.com/microsoft/AzureTRE/pull/TBD)
```
Comment (tamirkamara) [inline CHANGELOG.md:36]:
```suggestion
* API runs with gunicorn and uvicorn workers (as recommanded) [#3178](https://github.com/microsoft/AzureTRE/pull/3178)
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -33,6 +33,7 @@ ENHANCEMENTS:
 * Move non-core DNS zones out of the network module to reduce dependencies [#3119](https://github.com/microsoft/AzureTRE/pull/3119)
 * Review VMs are being cleaned up when an Airlock request is canceled ([#3130](https://github.com/microsoft/AzureTRE/pull/3130))
 * Sample queries to investigate logs of the core TRE applications ([#3151](https://github.com/microsoft/AzureTRE/pull/3151))
+* API runs with gunicorn and uvicorn workers (as recommanded) [#3178](https://github.com/microsoft/AzureTRE/pull/3178)
```
Comment (martinpeck) [inline CHANGELOG.md:36]: super-nit: typo `recommanded` should be `recommended`

Comment (tamirkamara) [inline CHANGELOG.md:36]:
```suggestion
* API runs with gunicorn and uvicorn workers (as recommended) [#3178](https://github.com/microsoft/AzureTRE/pull/3178)
```

**Pull request merged**